study-builder/src/main/java/org/broadinstitute/ddp/studybuilder/task/RarexVersion2.java

```diff
@@ -0,0 +1,723 @@
+        }
+    }
+
+    private void replaceQuestionBlockVariableText(long activityId, String stableId, String variableName, String text) {
+        List<Long> variableIds = helper.findQuestionBlockVariableIdsByVarNameAndStableId(activityId, variableName, stableId);
+        if (CollectionUtils.isEmpty(variableIds)) {
+            throw new DDPException("Could not find any variable with name " + variableName);
+        }
+        for (Long variableId : variableIds) {
+            helper.updateVarValueByTemplateVarId(variableId, text);
+            LOG.info("Updated question {} block variable {} in activity {} with text \"{}\"", stableId, variableId, activityId, text);
+        }
+    }
+
+    private void addPicklistOption(Handle handle, long studyId, Set<String> questionStableIds, Config option, int displayOrder) {
+        PicklistQuestionDao plQuestionDao = handle.attach(PicklistQuestionDao.class);
+        JdbiQuestion jdbiQuestion = handle.attach(JdbiQuestion.class);
+        Stream<QuestionDto> questionDtos = jdbiQuestion.findLatestDtosByStudyIdAndQuestionStableIds(studyId, questionStableIds);
+        questionDtos.forEach(questionDto -> {
+            PicklistOptionDef pickListOptionDef = gson.fromJson(ConfigUtil.toJson(option), PicklistOptionDef.class);
+            plQuestionDao.insertOption(questionDto.getId(), pickListOptionDef, displayOrder, questionDto.getRevisionId());
+            LOG.info("Added new picklistOption " + pickListOptionDef.getStableId() + " with id "
+                    + pickListOptionDef.getOptionId() + " into question " + questionDto.getStableId());
+        });
+    }
+
+    private void addValidation(Handle handle, long studyId, Collection<String> stableIds, Config ruleConfig) {
+        ValidationDao validationDao = handle.attach(ValidationDao.class);
+        for (String stableId : stableIds) {
+            QuestionDto questionDto = handle.attach(JdbiQuestion.class)
+                    .findLatestDtoByStudyIdAndQuestionStableId(studyId, stableId)
+                    .orElseThrow(() -> new DDPException("Could not find question " + stableId));
+            RequiredRuleDef rule = gson.fromJson(ConfigUtil.toJson(ruleConfig), RequiredRuleDef.class);
+            validationDao.insert(questionDto.getId(), rule, questionDto.getRevisionId());
+            LOG.info("Inserted validation rule with id={} questionStableId={}", rule.getRuleId(), questionDto.getStableId());
+        }
+    }
+
+    private FormActivityDef findActivityDef(String activityCode, String versionTag) {
+        ActivityDto activityDto = jdbiActivity
+                .findActivityByStudyIdAndCode(studyId, activityCode).get();
+        ActivityVersionDto versionDto = jdbiVersion
+                .findByActivityCodeAndVersionTag(studyId, activityCode, versionTag).get();
+        return (FormActivityDef) activityDao
+                .findDefByDtoAndVersion(activityDto, versionDto);
+    }
+
+    private interface SqlHelper extends SqlObject {
```

Comment:
There's some really helpful stuff here--it would be great to see this in its own class somewhere in our dao package.  Will be generally useful for incremental study changes!

Reply:
Yes, some of the methods could be reused. Let's do that outside of this PR